clang/include/clang/Basic/DiagnosticParseKinds.td

```diff
@@ -1713,6 +1713,12 @@ let CategoryName = "Reflection Issue" in {
 def err_cannot_reflect_entity : Error<"cannot reflect entity">;
 def err_expansion_statements_disabled : Error<
   "'template for' statements are not enabled; use '-fexpansion-statements'">;
+
+def warn_parsing_ambiguity_in_refl_expression_with_reference_types : Warning <
+    "given expression may be misinterpreted due to '%0 %1'"
+    " type parsing logic. did you mean '... (^%0) %1 ...'?"
```

Comment:
nit: maybe something like:

> `token '&&' binds left to the type 'int &&'; did you mean '(^int) &&'?`

Reply:
Changed. But I still cannot get name of type :(
So message looks like this: `token '&' binds left to type 'T &'; did you mean '(^T) &'?`

I checked similar places in a code to get an idea where to get type name but in all methods names are empty for my tests.